src/java.xml.crypto/share/classes/javax/xml/crypto/dsig/spec/RSAPSSParameterSpec.java

```diff
@@ -0,0 +1,145 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javax.xml.crypto.dsig.spec;
+
+import javax.xml.crypto.dsig.SignatureMethod;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.MGF1ParameterSpec;
+import java.security.spec.PSSParameterSpec;
+import java.util.Objects;
+
+/**
+ * Parameters for the <a href="http://www.w3.org/2007/05/xmldsig-more#rsa-pss">
+ * XML Signature RSASSA-PSS Algorithm</a>. The parameters are expressed as a
+ * {@link PSSParameterSpec} object encapsulated.
+ * <p>
+ * The XML Schema Definition is defined as:
+ * <pre><code>
+ *    &lt;xs:element name="RSAPSSParams" type="pss:RSAPSSParamsType"&gt;
+ *        &lt;xs:annotation&gt;
+ *            &lt;xs:documentation&gt;
+ *    Top level element that can be used in xs:any namespace="#other"
+ *    wildcard of ds:SignatureMethod content.
+ *            &lt;/xs:documentation&gt;
+ *        &lt;/xs:annotation&gt;
+ *    &lt;/xs:element&gt;
+ *    &lt;xs:complexType name="RSAPSSParamsType"&gt;
+ *        &lt;xs:sequence&gt;
+ *            &lt;xs:element ref="ds:DigestMethod" minOccurs="0"/&gt;
+ *            &lt;xs:element name="MaskGenerationFunction"
+ *               type="pss:MaskGenerationFunctionType" minOccurs="0"/&gt;
+ *            &lt;xs:element name="SaltLength" type="xs:int"
+ *               minOccurs="0"/&gt;
+ *            &lt;xs:element name="TrailerField" type="xs:int"
+ *               minOccurs="0"/&gt;
+ *        &lt;/xs:sequence&gt;
+ *    &lt;/xs:complexType&gt;
+ *    &lt;xs:complexType name="MaskGenerationFunctionType"&gt;
+ *        &lt;xs:sequence&gt;
+ *            &lt;xs:element ref="ds:DigestMethod" minOccurs="0"/&gt;
+ *        &lt;/xs:sequence&gt;
+ *        &lt;xs:attribute name="Algorithm" type="xs:anyURI"
+ *           default="http://www.w3.org/2007/05/xmldsig-more#MGF1"/&gt;
+ *    &lt;/xs:complexType&gt;
+ * </code></pre>
+ *
+ * @implSpec
+ * If the parameter is not specified when using the RSASSA-PSS signature
+ * algorithm, the default parameter is used, which uses SHA-256 as the
+ * {@code DigestMethod}, MGF1 with SHA-256 as the
+ * {@code MaskGenerationFunction}, 32 as {@code SaltLength}, and 1 as
+ * {@code TrailerField}. This is equivalent to the parameter-less signature
+ * method as defined by http://www.w3.org/2007/05/xmldsig-more#sha256-rsa-MGF1.
+ *
```

Comment:
Normally I don't like to hardcode defaults (in case they weaken later) but since this is specified by RFC 6931, I don't think we have a choice, and I think we need to use this default for interoperability.